File: Jessica/121.Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices)<2:
            return 0
        MinP = prices[0]
        MaxProfit=0
        for i in range(1,len(prices)):
            if prices[i] > MinP:
                MaxProfit=max(MaxProfit,prices[i] - MinP)
            else:
                MinP=min(MinP,prices[i])
                
        return MaxProfit
        
           

File: Jessica/121.Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()<2){
            return 0;
        }
        int low = prices[0];
        int MaxProfit=0;
        for(int i=1;i<prices.size();i++){
            if (prices[i]>low){
                MaxProfit=max(MaxProfit,prices[i]-low);                
            }
            else{
                low=min(low,prices[i]);
            }
        }
        
        return MaxProfit;
    }
};
